src/conn/conn_log.c

```diff
@@ -79,28 +79,23 @@ __logmgr_force_archive(WT_SESSION_IMPL *session, uint32_t lognum)
 }
 
 /*
- * __logmgr_set_majmin --
- *     Set the required major or minor of the given field. Wrapper for setting the required minimum
- *     and maximum fields in the connection.
+ * __logmgr_set_log_version --
+ *     Set the log version required for the given WiredTiger version.
  */
 static void
-__logmgr_set_majmin(uint16_t req_major, uint16_t req_minor, uint16_t *log_req)
+__logmgr_set_log_version(WT_VERSION version, uint16_t *log_req)
 {
-    /*
-     * Set up the maximum and minimum log version required if needed.
-     */
-    if (req_major != WT_CONN_COMPAT_NONE) {
-        if (req_major == WT_LOG_V5_MAJOR)
-            *log_req = WT_LOG_VERSION;
-        else if (req_major == WT_LOG_V4_MAJOR)
-            if (req_minor == WT_LOG_V4_MINOR)
-                *log_req = 4;
-            else if (req_minor > WT_LOG_V2_MINOR)
-                *log_req = 3;
-            else
-                *log_req = 2;
-        else
+    if (__wt_version_defined(version)) {
```

Comment:
Should we still set `log_req` if this is `false`? The header says the function sets a value to `log_req`

Reply:
Good point. Updated in b683cf8.

I originally left this out to match the original behaviour where in the callstack
```
wiredtiger_open
    __conn_version_verify
        __wt_logmgr_compat_version
            __logmgr_get_log_version
```
`version` (in this case `conn->compat_req_min` and `conn->compat_req_max`) are not defined meaning we don't want to set `log_req` (`conn->log_req_min` and `conn->log_req_max`).

However `__logmgr_get_log_version` is the only location in the code in which the values of `log_req_min`/`log_req_min` are set, indicating that when  `__wt_version_defined(version)` was false we were leaving these values uninitialised. Setting them to a default `NO_VALUE` instead seems cleaner.